### ISSUE TITLE
Use flags instead of strings to parse user defined parameters

### ISSUE THREAD

**Issue by hackermd** [TissueMAPS/TissueMAPS#87]
String parameters provided by users (e.g. via workflow YAML description) should not be parsed directly to the constructor of the respective class in form of a string, but rather handled via flags (similar to C++ `enum` type).

**Comment by riccardomurri** [comment]
I'm not sure what is the suggestion here... what is a "flag" for you?  where/how should string parameters be parsed? (C and C++ do not have anything called "flag" either: the names in an enumeration are "identifiers", see e.g. http://en.cppreference.com/w/cpp/language/enum )

**Comment by hackermd** [comment]
Yes, I am referring to enumeration "identifiers". OpenCV calls them flags, e.g. https://docs.opencv.org/3.2.0/d4/d15/group__videoio__flags__base.html

**Comment by riccardomurri** [comment]
Still I'm not sure what exactly you are suggesting... They need to be entered as strings anyway (Python does not have a separate identifier/symbol type), and be converted internally to what?  And where do you envision this would happen?

**Comment by hackermd** [comment]
Take for example the `mode` parameter, which is parsed to the constructor of `WorkflowStageDescription` (see [here](https://github.com/TissueMAPS/TmLibrary/blob/a0006ad63e468c9717dd1309e15005fcb6084a49/tmlib/workflow/description.py#L168)). The argument must be a string and can be either `"sequential"` or `"parallel"`.

Sure users can only provide the value in form of a string via the YAML document. However, in Python this could be more elegantly represented as a static class that maps these string identifiers to numerical values using class attributes:

```python
class Mode(object):

    SEQUENTIAL = 0
    PARALLEL = 1
```

Now the argument for the constructor would be either the `Mode.SEQUENTIAL` or `Mode.PARALLEL` "flag".

This would simplify checking of user input and error handling:

```python
mode = 'sequential' # provided by user
try:
    val = getattr(Mode, mode.upper())
except AttributeError:
    raise ValueError('Invalid mode "{}"'.format(mode))
```

and would facilitate the generation of a list of available options for clients to select one of them, e.g. via a drop down menu:

```python
[v.lower() for v in vars(Mode) if not v.startswith('_')]
```
